core/src/main/java/org/elasticsearch/index/translog/Checkpoint.java

```diff
@@ -97,18 +97,21 @@ private void write(DataOutput out) throws IOException {
     }
 
     static Checkpoint emptyTranslogCheckpoint(final long offset, final long generation, final long globalCheckpoint) {
-        return new Checkpoint(offset, 0, generation, Translog.INITIAL_MIN_SEQ_NO, Translog.INITIAL_MAX_SEQ_NO, globalCheckpoint);
+        final long minSeqNo = SequenceNumbersService.NO_OPS_PERFORMED;
+        final long maxSeqNo = SequenceNumbersService.NO_OPS_PERFORMED;
+        return new Checkpoint(offset, 0, generation, minSeqNo, maxSeqNo, globalCheckpoint);
     }
 
-    static Checkpoint readChecksummedV2(DataInput in) throws IOException {
+    static Checkpoint readChecksummedV2(final DataInput in) throws IOException {
         return new Checkpoint(in.readLong(), in.readInt(), in.readLong(), in.readLong(), in.readLong(), in.readLong());
     }
 
     // reads a checksummed checkpoint introduced in ES 5.0.0
-    static Checkpoint readChecksummedV1(DataInput in) throws IOException {
-        final long minSeqNo = Translog.INITIAL_MIN_SEQ_NO;
-        final long maxSeqNo = Translog.INITIAL_MAX_SEQ_NO;
-        return new Checkpoint(in.readLong(), in.readInt(), in.readLong(), minSeqNo, maxSeqNo, SequenceNumbersService.UNASSIGNED_SEQ_NO);
+    static Checkpoint readChecksummedV1(final DataInput in) throws IOException {
+        final long minSeqNo = SequenceNumbersService.NO_OPS_PERFORMED;
```

Comment:
these should be UNASSIGNED_SEQ_NO no? (unless maybe if total count is 0, but I'm not sure it's worth the code song and dance it will require)